### PR TITLE
app/vmagent: adds pubsub integration for enterprise version

### PR DESCRIPTION
**Pull request by f41gh7**

### CODE REVIEW

File: app/vmagent/README.md [hunk 1]
```diff
@@ -1029,6 +1029,124 @@ If you have suggestions for improvements or have found a bug - please open an is
...
 See also [troubleshooting docs](https://docs.victoriametrics.com/Troubleshooting.html).
 
+## Google PubSub integration
+[Enterprise version](https://docs.victoriametrics.com/enterprise.html) of `vmagent` can read and write metrics from / to google [PubSub](https://cloud.google.com/pubsub):
+
+
+### Reading metrics from PubSub
+
+[Enterprise version](https://docs.victoriametrics.com/enterprise.html) of `vmagent` can read metrics in various formats from Pub/Sub messages.
+These formats can be configured with `-gcp.pubsub.subscribe.defaultMessageFormat` or `-gcp.pubsub.subscribe.topicSubscription.messageFormat` command-line options. The following formats are supported:
```
Comment (tenmozes) [inline app/vmagent/README.md:1039]: `-gcp.pubsub.subscribe.defaultMessageFormat` and `-gcp.pubsub.subscribe.topicSubscription.messageFormat` allow to configure the message format 
We support the following options:

File: app/vmagent/README.md [hunk 2]
```diff
@@ -1029,6 +1029,124 @@ If you have suggestions for improvements or have found a bug - please open an is
...
+* `influx` - [InfluxDB line protocol format](https://docs.influxdata.com/influxdb/v1.7/write_protocols/line_protocol_tutorial/).
+* `prometheus` - [Prometheus text exposition format](https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md#text-based-format)
+  and [OpenMetrics format](https://github.com/OpenObservability/OpenMetrics/blob/master/specification/OpenMetrics.md).
+* `graphite` - [Graphite plaintext format](https://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-plaintext-protocol).
+* `jsonline` - [JSON line format](https://docs.victoriametrics.com/#how-to-import-data-in-json-line-format).
+
+Every PubSub message may contain multiple lines in `influx`, `prometheus`, `graphite` and `jsonline` format delimited by `\n`.
+
+`vmagent` consumes messages from PubSub topic subscriptions specified by `-gcp.pubsub.subscribe.topicSubscription` command-line flag. Multiple topics can be specified
+by passing multiple `-gcp.pubsub.subscribe.topicSubscription` command-line flags to `vmagent`.
```
Comment (tenmozes) [inline app/vmagent/README.md:1052]: You can configure the multiple topics by specifying `-gcp.pubsub.subscribe.topicSubscription` command-line flags in `vmagent`

File: app/vmagent/README.md [hunk 3]
```diff
@@ -1029,6 +1029,124 @@ If you have suggestions for improvements or have found a bug - please open an is
...
+
+`vmagent` uses standard google authorization mechanism for topic access. It's possible to specify credentials directly via flag `-gcp.pubsub.subscribe.credentialsFile=`.
```
Comment (tenmozes) [inline app/vmagent/README.md:1054]: `vmagent` uses a standard Google authorization mechanism for topic access. It's possible to specify credentials directly via the flag `-gcp.pubsub.subscribe.credentialsFile=`.

File: app/vmagent/README.md [hunk 4]
```diff
@@ -1029,6 +1029,124 @@ If you have suggestions for improvements or have found a bug - please open an is
...
+It is expected that [Telegraf](https://github.com/influxdata/telegraf) sends metrics to the `metrics-by-telegraf` topic with the following config:
+
+```yaml
+[[outputs.cloud_pubsub]]
+  project = "victoriametrics-vmagent-pub-sub-test"
+  topic = "telegraf-testing"
+  data_format = "influx"
+```
+
+#### Consume from multiple topics
```
Comment (tenmozes) [inline app/vmagent/README.md:1074]: Consume metrics from...

File: app/vmagent/README.md [hunk 5]
```diff
@@ -1029,6 +1029,124 @@ If you have suggestions for improvements or have found a bug - please open an is
...
+
+It's possible to configure message consumption with multiple topics. In this case, command-line flag arguments must have the same position at corresponding values.
```
Comment (tenmozes) [inline app/vmagent/README.md:1076]: consumption from multiple topics.

**Review by tenmozes — CHANGES_REQUESTED** [review]

**Comment by valyala** [comment]
@f41gh7 , thanks for the docs!

**Pull request merged**